data_steward/retraction/retract_utils.py

```diff
@@ -134,6 +135,36 @@ def get_datasets_list(project_id, dataset_ids_str):
     return dataset_ids
 
 
+def is_deid_label_or_id(client, project_id, dataset_id):
+    """
+    Validates if a dataset is labeled deid or contains deid in the dataset_id
+
+    :param client: BigQuery client
+    :param project_id: project containing the dataset
+    :param dataset_id: dataset to identify
+    :return: Boolean indicating if a dataset is a deid dataset
+    """
+    label = is_labeled_deid(client, project_id, dataset_id)
+    if label is None:
+        return is_deid_dataset(dataset_id)
+    return label
+
+
+def is_labeled_deid(client, project_id, dataset_id):
+    """
+    Returns boolean indicating if a dataset is a deid dataset using the label 'de_identified'
+
+    :param client: BigQuery client object
+    :param project_id: Identifies the project
+    :param dataset_id: Identifies the dataset
+    :return: Boolean indicating if the dataset is labeled a deid dataset or None if unlabeled
+    """
+    dataset = client.get_dataset(f'{project_id}.{dataset_id}')
+    if dataset.labels and consts.DE_IDENTIFIED in dataset.labels:
+        return dataset.labels[consts.DE_IDENTIFIED] == consts.TRUE
```

Comment:
optional:  consider checking the lower cased value.  So that `True` would also be accepted.

How input proof should we make this?  For example, we want people to use `true`, but what if the input is `T` or `trUE` or `Y` or `YES`.  We could check for lowercase variants of these with a list, but what if someone put in `of course` or some other random string value?  Should we raise an error here if the value isn't set to an accepted true or false value?

Reply:
I think an error should be raised, but ideally where the label is set rather than here. BQ will raise the error if the label value contains a letter in caps while assigning.

Reply:
Good point!  I forgot about that case label restriction.  But what do you think about the other idea?  Should this check for only `true` or should it look for any common positive response ['true', 't', 'yes', 'y']?  Thoughts?

I know we've only used 'true' before now.  I'm just curious what you think about this.

Reply:
I think we should just stick to one value to indicate a certain characteristic of the dataset to reduce ambiguity. This will help standardize it to a convention that can be enforced while assigning.

Reply:
Sounds good.  Just wanted to bring it up for your consideration.